<a href="https://colab.research.google.com/github/HIsu1231/CUKNLP_INSTAHASHTAG/blob/main/Final_Dataset%EB%A7%8C%EB%93%A4%EA%B8%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import os

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
path = '/content/drive/MyDrive/NLP연구실 공유문서함/원현식/인스타/hashtag/'
file_list = os.listdir(path)
print(file_list)
#['love', 'nofilter', 'photography', 'happy', 'smile', 'model', 'summer', 'cute', 'beauty', 'nature', 'dog', 'instagram', 'music', 'beach', 'sunset', 'style', 'life', 'instadaily', 'photo', 'food', 'fitness', 'family', 'motivation', 'friends', 'amazing', 'makeup', 'foodporn', 'me', 'instamood', 'ootd', 'picoftheday', 'design', 'hair', 'flowers', 'cat', 'vscocam', 'sky', 'bestoftheday', 'vsco', 'artist', 'drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle', 'london', 'black', 'home', 'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'daily', 'pretty', 'nyc', 'explore', 'baby', 'landscape']
#['love', 'nofilter', 'photography', 'happy', 'smile', 'model', 'summer', 'cute', 'beauty', 'nature', 'dog', 'instagram', 'music', 'beach', 'sunset', 'style', 'life', 'instadaily', 'photo', 'food', 'fitness', 'family', 'motivation', 'friends', 'amazing', 'makeup', 'foodporn', 'me', 'instamood', 'ootd', 'picoftheday', 'design', 'hair', 'flowers', 'cat', 'vscocam', 'sky', 'bestoftheday', 'vsco', 'artist', 'drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle', 'london', 'black', 'home', 'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'pretty', 'nyc', 'explore', 'baby', 'landscape', 'funny', 'fashion', 'catsofinstagram', 'travel', 'beautiful', 'nails', 'art', 'igers', 'selfie', 'photooftheday', 'japan', 'fun', 'gym', 'moda', 'workout', 'inspiration', 'fit', 'weekend', 'winter', 'daily']

['love', 'nofilter', 'photography', 'happy', 'smile', 'model', 'summer', 'cute', 'beauty', 'nature', 'dog', 'instagram', 'music', 'beach', 'sunset', 'style', 'life', 'instadaily', 'photo', 'food', 'fitness', 'family', 'motivation', 'friends', 'amazing', 'makeup', 'foodporn', 'me', 'instamood', 'ootd', 'picoftheday', 'design', 'hair', 'flowers', 'cat', 'vscocam', 'sky', 'bestoftheday', 'vsco', 'artist', 'drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle', 'london', 'black', 'home', 'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'pretty', 'nyc', 'explore', 'baby', 'landscape', 'funny', 'fashion', 'catsofinstagram', 'travel', 'beautiful', 'nails', 'art', 'igers', 'selfie', 'photooftheday', 'japan', 'fun', 'gym', 'moda', 'workout', 'inspi

In [ ]:
path = '/content/drive/MyDrive/NLP연구실 공유문서함/원현식/인스타/hashtag/'
file_list = os.listdir(path)
print(file_list)
#['love', 'nofilter', 'photography', 'happy', 'smile', 'model', 'summer', 'cute', 'beauty', 'nature', 'dog', 'instagram', 'music', 'beach', 'sunset', 'style', 'life', 'instadaily', 'photo', 'food', 'fitness', 'family', 'motivation', 'friends', 'amazing', 'makeup', 'foodporn', 'me', 'instamood', 'ootd', 'picoftheday', 'design', 'hair', 'flowers', 'cat', 'vscocam', 'sky', 'bestoftheday', 'vsco', 'artist', 'drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle', 'london', 'black', 'home', 'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'daily', 'pretty', 'nyc', 'explore', 'baby', 'landscape']

['love', 'nofilter', 'photography', 'happy', 'smile', 'model', 'summer', 'cute', 'beauty', 'nature', 'dog', 'instagram', 'music', 'beach', 'sunset', 'style', 'life', 'instadaily', 'photo', 'food', 'fitness', 'family', 'motivation', 'friends', 'amazing', 'makeup', 'foodporn', 'me', 'instamood', 'ootd', 'picoftheday', 'design', 'hair', 'flowers', 'cat', 'vscocam', 'sky', 'bestoftheday', 'vsco', 'artist', 'drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle', 'london', 'black', 'home', 'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'pretty', 'nyc', 'explore', 'baby', 'landscape', 'funny', 'fashion', 'catsofinstagram', 'travel', 'beautiful', 'nails', 'art', 'igers', 'selfie', 'photooftheday', 'japan', 'fun', 'gym', 'moda', 'workout', 'inspi

In [ ]:
for i in file_list:
  df = pd.read_csv(path + i + '/' + i + '-change2.csv', index_col=0)
  df.to_csv(path + i + '/' + i + '-change2_drop.csv')
  df = pd.read_csv(path + i + '/' + i + '-change2_drop.csv', index_col=0)
  df 

  index = []
  for l in range(len(df)):
    df['filename'] = i
    if df['after_contents'][l] == ' ':
      index.append(l)
    if str(df['after_contents'][l]) == 'nan':
      index.append(l)
    if df['images'][l] == '[]':
      index.append(l)

  df = df.drop(index, axis=0)
  df = df.reset_index()

  df['drop'] = ''
  for j in range(len(index)):
    df['drop'][j] = index[j]
  
  df.to_csv(path + i + '/' + i + '-change2_drop.csv')

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:22: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy


In [ ]:
path ='./content/drive/MyDrive/인스타/hashtag'

In [ ]:
import pandas as pd

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/인스타/hashtag/drawing/drawing-change2.csv', index_col=0)
df.to_csv('/content/drive/MyDrive/인스타/hashtag/drawing/drawing-change2_drop.csv')
df = pd.read_csv('/content/drive/MyDrive/인스타/hashtag/drawing/drawing-change2_drop.csv', index_col=0)
df 

index = []
for l in range(len(df)):
  df['filename'] = 'home'
  if df['after_contents'][l] == ' ':
    index.append(l)
  if str(df['after_contents'][l]) == 'nan':
    index.append(l)
  if df['images'][l] == '[]':
    index.append(l)

df = df.drop(index, axis=0)
df = df.reset_index()

df['drop'] = ''
for j in range(len(index)):
  df['drop'][j] = index[j]

df.to_csv('/content/drive/MyDrive/인스타/hashtag/drawing/drawing-change2_drop.csv')

FileNotFoundError: ignored

In [ ]:
df = pd.read_csv(path + file_list[1] + '/' + file_list[1] + '-change2_drop.csv', index_col=0)
df

index            upload_id                 date_time  \
0        2         trishy_cakes  2022-04-02T21:55:33.000Z   
1        3       necro.manteion  2022-04-02T20:39:52.000Z   
2        5  teppei.tepi.okumura  2022-04-03T08:48:10.000Z   
3        7      rahul__dasadiya  2022-04-03T08:47:25.000Z   
4        8        gianlucategas  2022-04-03T08:47:09.000Z   
..     ...                  ...                       ...   
287    404            meljayne1  2022-04-03T08:02:54.000Z   
288    405  bedacmlodymdentysta  2022-04-03T08:02:12.000Z   
289    406            anjafidux  2022-04-03T08:02:26.000Z   
290    407      stefanoprincipi  2022-04-02T21:50:51.000Z   
291    408               ljim91  2022-04-03T08:02:58.000Z   

                                              hashtags  \
0    ['envecaliforniaaesthetics', 'EnVe', 'alastins...   
1    ['caturday', 'chatombale', 'perelachaisecemete...   
2    ['hongkong', 'taimeituk', 'taipo', 'sunset', '...   
3                                         ['nofilter']   
4    ['nofilter', 'paradise', 'sardegna', 'sunset',...   
..                                                 ...   
287  ['sunset', 'nofilter', 'sunsetsandsunrises', '...   
288  ['april', 'winterintospring', 'recesivegene', ...   
289  ['aprillove', 'aprillove2022', 'sanlorenzo', '...   
290                        ['neve', 'terni', 'umbria']   
291  ['instagood', 'instamood', 'instadaily', 'inst...   

                                              contents  \
0    trishy_cakesFreckles out✨ Loving my skin ...#e...   
1    necro.manteionIt's #caturday ! #chatombale fro...   
2    teppei.tepi.okumura.well, just go with the flo...   
3                   rahul__dasadiyaSmile😎 #nofilter17분   
4    gianlucategasHome...#nofilter #paradise #sarde...   
..                                                 ...   
287  meljayne1What an amazing sunset!! 🌄😍...#sunset...   
288  bedacmlodymdentystaWinter into spring :) Babe,...   
289  anjafidux01. Blue#aprillove #aprillove2022 #sa...   
290  stefanoprincipiUna magica notte di primavera.#...   
291  ljim91Cardiff Half Marathon 27.03.2022.This wa...   

                                                images  \
0    ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
1    ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
2    ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
3    ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
4    ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
..                                                 ...   
287  ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
288  ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
289  ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
290  ['https://scontent-gmp1-1.cdninstagram.com/v/t...   
291  ['https://scontent-gmp1-1.cdninstagram.com/v/t...   

                                    location_info  \
0                      EnVē California Aesthetics   
1                      Cimetière du Père-Lachaise   
2                                     Tai Mei Tuk   
3                                 शंखेश्वर गुजरात   
4                         Poetto, Sardegna, Italy   
..                                            ...   
287                                Lake Macquarie   
288                                           NaN   
289                                           NaN   
290                      Cecalocco, Umbria, Italy   
291  Cardiff City Hall / Neuadd y Ddinas Caerdydd   

                                         location_href  \
0    https://www.instagram.com/explore/locations/10...   
1    https://www.instagram.com/explore/locations/23...   
2    https://www.instagram.com/explore/locations/10...   
3    https://www.instagram.com/explore/locations/16...   
4    https://www.instagram.com/explore/locations/21...   
..                                                 ...   
287  https://www.instagram.com/explore/locations/15...   
288                                                NaN   
289                   

In [ ]:
df1 = pd.read_csv(path + file_list[0] + '/' + file_list[0] + '-change2_drop.csv', index_col=0, encoding='utf-8-sig')
df1 = df1[['index', 'hashtags', 'after_contents', 'new_hashtags', 'new_hashtags_onehot', 'filename', 'drop']]
for i in range(1, len(file_list)):
  df2 = pd.read_csv(path + file_list[i] + '/' + file_list[i] + '-change2_drop.csv', index_col=0, encoding='utf-8-sig')
  df2 = df2[['index', 'hashtags', 'after_contents', 'new_hashtags', 'new_hashtags_onehot', 'filename', 'drop']]
  df1 = pd.concat([df1, df2], )
  df1.reset_index(drop=True, inplace=True)

df1  

index                                           hashtags  \
0          0  ['poetry', 'quote', 'indigenous', 'meditation'...   
1          2  ['polymerclay', 'artistsofinstagram', 'miniatu...   
2          3      ['healing', 'spirituality', 'growth', 'love']   
3          4  ['baby', 'babiesofinstagram', 'firstbirthday',...   
4          6                             ['new', 'love', 'dio']   
...      ...                                                ...   
22828     84  ['messej', 'letsmessej', 'messejtips', 'social...   
22829     87                                                 []   
22830     88  ['foryou', 'foryourpage', 'daily', 'viral', 'p...   
22831     89  ['angles', 'oraclecommunity', 'oreclereaderofi...   
22832     90                                                 []   

                                          after_contents  \
0      love frees you into peace returning you to you...   
1                  1 chef salad left should i make more    
2      today we had a very tough conversation in our ...   
3      happiest first birthday baby boy love you sooo...   
4      life s way too short for bad vibes post master...   
...                                                  ...   
22828  start messaging with your dear ones in your re...   
22829                                                 1    
22830  first time mili sir so humble and nice person ...   
22831  this is a general reading take what resonates ...   
22832                                                 6    

                                            new_hashtags  \
0      [336, 343, 634, 92, 780, 448, 778, 903, 964, 9...   
1                    [147, 50, 363, 125, 45, 8, 4, 0, 1]   
2                                          [780, 946, 0]   
3                                     [123, 794, 272, 0]   
4                                               [223, 0]   
...                                                  ...   
22828     [242, 13, 3, 57, 48, 52, 46, 1, 526, 562, 755]   
22829                                                 []   
22830                                [186, 242, 48, 857]   
22831                                     [634, 242, 32]   
22832                                                 []   

                                     new_hashtags_onehot filename  drop  
0      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love   1.0  
1      [1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, ...     love   5.0  
2      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love   8.0  
3      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love  12.0  
4      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love  20.0  
...                                                  ...      ...   ...  
22828  [0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, ...    daily  68.0  
22829  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  71.0  
22830  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  76.0  
22831  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  79.0  
22832  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  81.0  

[22833 rows x 7 columns]

In [ ]:
df1.to_csv(path + 'all.csv', encoding='utf-8-sig')

In [ ]:
sum = 0
for i in range(0, len(file_list)):
  df = pd.read_csv(path + file_list[i] + '/' + file_list[i] + '-change2_drop.csv', index_col=0)
  sum += len(df)
sum

24783

In [ ]:
df1

index                                           hashtags  \
0          0  ['poetry', 'quote', 'indigenous', 'meditation'...   
1          2  ['polymerclay', 'artistsofinstagram', 'miniatu...   
2          3      ['healing', 'spirituality', 'growth', 'love']   
3          4  ['baby', 'babiesofinstagram', 'firstbirthday',...   
4          6                             ['new', 'love', 'dio']   
...      ...                                                ...   
22828     84  ['messej', 'letsmessej', 'messejtips', 'social...   
22829     87                                                 []   
22830     88  ['foryou', 'foryourpage', 'daily', 'viral', 'p...   
22831     89  ['angles', 'oraclecommunity', 'oreclereaderofi...   
22832     90                                                 []   

                                          after_contents  \
0      love frees you into peace returning you to you...   
1                  1 chef salad left should i make more    
2      today we had a very tough conversation in our ...   
3      happiest first birthday baby boy love you sooo...   
4      life s way too short for bad vibes post master...   
...                                                  ...   
22828  start messaging with your dear ones in your re...   
22829                                                 1    
22830  first time mili sir so humble and nice person ...   
22831  this is a general reading take what resonates ...   
22832                                                 6    

                                            new_hashtags  \
0      [336, 343, 634, 92, 780, 448, 778, 903, 964, 9...   
1                    [147, 50, 363, 125, 45, 8, 4, 0, 1]   
2                                          [780, 946, 0]   
3                                     [123, 794, 272, 0]   
4                                               [223, 0]   
...                                                  ...   
22828     [242, 13, 3, 57, 48, 52, 46, 1, 526, 562, 755]   
22829                                                 []   
22830                                [186, 242, 48, 857]   
22831                                     [634, 242, 32]   
22832                                                 []   

                                     new_hashtags_onehot filename  drop  
0      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love   1.0  
1      [1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, ...     love   5.0  
2      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love   8.0  
3      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love  12.0  
4      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love  20.0  
...                                                  ...      ...   ...  
22828  [0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, ...    daily  68.0  
22829  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  71.0  
22830  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  76.0  
22831  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  79.0  
22832  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  81.0  

[22833 rows x 7 columns]

#LONDON,Home해라

In [1]:
import pandas as pd
import cv2
import os

In [3]:
path = '/content/drive/MyDrive/인스타/hashtag/'

In [ ]:
#london, home 해야함
file_list = ['drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle',  'black',  'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'pretty', 'nyc', 'explore', 'baby', 'landscape', 'funny', 'fashion', 'catsofinstagram', 'travel', 'beautiful', 'nails', 'art', 'igers', 'selfie', 'photooftheday', 'japan', 'fun', 'gym', 'moda', 'workout', 'inspiration', 'fit', 'weekend', 'winter', 'daily']

In [ ]:
for i in range(len(file_list)):
  df = pd.read_csv(path + file_list[i] + '/' + file_list[i] + '-change2_drop.csv', index_col=0, encoding='utf-8')
  print(file_list[i])
  for j in range(len(df)):
    index = df['index'][j]
    print('index: ', index, 'i:', j)
    image = cv2.imread(path + file_list[i] + '/' + str(index) +'-0.jpg' )
    filepath = path + 'image/' + file_list[i] + '/'+ str(j) +'.jpg'
    cv2.imwrite(filepath, image)




스트리밍 출력 내용이 길어서 마지막 5000줄이 삭제되었습니다.
index:  106 i: 86
index:  107 i: 87
index:  109 i: 88
index:  110 i: 89
index:  111 i: 90
index:  113 i: 91
index:  114 i: 92
index:  115 i: 93
index:  116 i: 94
index:  117 i: 95
index:  118 i: 96
index:  120 i: 97
index:  121 i: 98
index:  122 i: 99
index:  123 i: 100
index:  124 i: 101
index:  125 i: 102
index:  127 i: 103
index:  128 i: 104
index:  129 i: 105
index:  130 i: 106
index:  131 i: 107
index:  132 i: 108
index:  134 i: 109
index:  135 i: 110
index:  137 i: 111
index:  138 i: 112
index:  139 i: 113
index:  140 i: 114
index:  141 i: 115
index:  142 i: 116
index:  145 i: 117
index:  146 i: 118
index:  147 i: 119
index:  148 i: 120
index:  151 i: 121
index:  152 i: 122
index:  153 i: 123
index:  154 i: 124
index:  156 i: 125
index:  157 i: 126
index:  158 i: 127
index:  159 i: 128
index:  160 i: 129
index:  161 i: 130
index:  162 i: 131
index:  164 i: 132
index:  165 i: 133
index:  166 i: 134
index:  167 i: 135
index:  168 i: 136
index:  1

In [4]:
all = pd.read_csv(path + 'all_.csv', index_col=0)
all

index                                           hashtags  \
0          0  ['poetry', 'quote', 'indigenous', 'meditation'...   
1          2  ['polymerclay', 'artistsofinstagram', 'miniatu...   
2          3      ['healing', 'spirituality', 'growth', 'love']   
3          4  ['baby', 'babiesofinstagram', 'firstbirthday',...   
4          6                             ['new', 'love', 'dio']   
...      ...                                                ...   
22828     84  ['messej', 'letsmessej', 'messejtips', 'social...   
22829     87                                                 []   
22830     88  ['foryou', 'foryourpage', 'daily', 'viral', 'p...   
22831     89  ['angles', 'oraclecommunity', 'oreclereaderofi...   
22832     90                                                 []   

                                          after_contents  \
0      love frees you into peace returning you to you...   
1                  1 chef salad left should i make more    
2      today we had a very tough conversation in our ...   
3      happiest first birthday baby boy love you sooo...   
4      life s way too short for bad vibes post master...   
...                                                  ...   
22828  start messaging with your dear ones in your re...   
22829                                                 1    
22830  first time mili sir so humble and nice person ...   
22831  this is a general reading take what resonates ...   
22832                                                 6    

                                            new_hashtags  \
0      [336, 343, 634, 92, 780, 448, 778, 903, 964, 9...   
1                    [147, 50, 363, 125, 45, 8, 4, 0, 1]   
2                                          [780, 946, 0]   
3                                     [123, 794, 272, 0]   
4                                               [223, 0]   
...                                                  ...   
22828     [242, 13, 3, 57, 48, 52, 46, 1, 526, 562, 755]   
22829                                                 []   
22830                                [186, 242, 48, 857]   
22831                                     [634, 242, 32]   
22832                                                 []   

                                     new_hashtags_onehot filename  drop  
0      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love   1.0  
1      [1, 1, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, ...     love   5.0  
2      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love   8.0  
3      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love  12.0  
4      [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...     love  20.0  
...                                                  ...      ...   ...  
22828  [0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, ...    daily  68.0  
22829  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  71.0  
22830  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  76.0  
22831  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  79.0  
22832  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    daily  81.0  

[22833 rows x 7 columns]

In [58]:
london_index = []
for i in range(len(all)):
    if all['filename'][i] == 'london':
      london_index.append(all['index'][i])

In [59]:
london_index = list(map(int, london_index))

In [63]:
print(london_index)

[0, 2, 3, 4, 5, 6, 7, 8, 9, 11, 13, 14, 15, 16, 18, 19, 20, 21, 22, 23, 24, 25, 26, 28, 29, 31, 32, 33, 34, 35, 36, 37, 38, 39, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 52, 53, 54, 55, 57, 58, 60, 62, 65, 66, 67, 68, 70, 72, 74, 76, 77, 78, 80, 82, 88, 89, 90, 92, 93, 95, 96, 98, 99, 101, 103, 104, 107, 109, 110, 113, 114, 115, 116, 119, 121, 125, 126, 128, 129, 132, 133, 134, 135, 137, 138, 139, 140, 141, 142, 145, 146, 147, 148, 149, 150, 152, 153, 154, 156, 157, 158, 159, 160, 164, 165, 167, 170, 171, 174, 176, 178, 179, 180, 181, 182, 185, 186, 187, 188, 189, 191, 193, 195, 196, 197, 198, 199, 200, 201, 203, 205, 206, 207, 208, 210, 211, 212, 214, 215, 216, 217, 219, 220, 221, 225, 226, 227, 228, 229, 233, 234, 235, 236, 237, 239, 240, 241, 242, 243, 244, 245, 246, 247, 248, 249, 251, 252, 254, 255, 256, 257, 258, 260, 261, 262, 266, 267, 268, 270, 271, 272, 273, 274, 277, 278, 279, 280, 281, 282, 283, 284, 285, 287, 288, 289, 290, 294, 296, 299, 301, 302, 303, 305, 306, 307, 308, 3

In [72]:
london_index = [0, 11, 13, 14, 15, 101, 103, 104, 107, 109, 110, 113, 114, 115, 116, 119, 121, 125, 126, 128, 129, 132, 133, 134, 135, 137, 138, 139, 140, 141, 142, 145, 146, 147, 148, 149, 150, 152, 153, 154, 156, 157, 158, 159, 160, 164, 165, 167, 170, 171, 174, 178,  182, 185, 186, 187, 208, 214, 227, 241, 266, 282, 283, 284, 329]

In [70]:
path

'/content/drive/MyDrive/NLP연구실 공유문서함/원현식/인스타/hashtag/'

In [73]:
path = '/content/drive/MyDrive/NLP연구실 공유문서함/원현식/인스타/hashtag/'
for j in london_index:
  image = cv2.imread(path + 'london/' + str(j) +'-0.jpg' )
  filepath = path + 'image/london/'+ str(j) +'.jpg'
  cv2.imwrite(filepath, image)
  print(j)


0
11
13
14
15
101
103
104
107
109
110
113
114
115
116
119
121
125
126
128
129
132
133
134
135
137
138
139
140
141
142
145
146
147
148
149
150
152
153
154
156
157
158
159
160
164
165
167
170
171
174
178
182
185
186
187
208
214
227
241
266
282
283
284
329


##전체 이미지 파일 한 폴더에 인덱싱

In [7]:
path = '/content/drive/MyDrive/인스타/hashtag/image'
file_list = os.listdir(path)
print(file_list)

['love', 'nofilter', 'photography', 'happy', 'smile', 'model', 'summer', 'cute', 'beauty', 'nature', 'dog', 'instagram', 'music', 'beach', 'sunset', 'style', 'life', 'instadaily', 'photo', 'food', 'fitness', 'family', 'motivation', 'friends', 'amazing', 'makeup', 'foodporn', 'me', 'instamood', 'ootd', 'picoftheday', 'design', 'hair', 'flowers', 'cat', 'vscocam', 'sky', 'bestoftheday', 'vsco', 'artist', 'drawing', 'girl', 'dogsofinstagram', 'handmade', 'instafood', 'photographer', 'wedding', 'party', 'girls', 'healthy', 'lol', 'swag', 'cool', 'foodie', 'blackandwhite', 'lifestyle', 'london', 'black', 'christmas', 'instapic', 'memes', 'holiday', 'sea', 'blue', 'pink', 'night', 'puppy', 'illustration', 'work', 'instafashion', 'yummy', 'happiness', 'architecture', 'tattoo', 'pretty', 'nyc', 'explore', 'baby', 'landscape', 'funny', 'fashion', 'catsofinstagram', 'travel', 'beautiful', 'nails', 'art', 'igers', 'selfie', 'photooftheday', 'japan', 'fun', 'gym', 'moda', 'workout', 'inspiration',

In [8]:
file_list.sort()
file_list = file_list[1:]
print(len(file_list))

100


In [9]:
file_list

['amazing',
 'architecture',
 'art',
 'artist',
 'baby',
 'beach',
 'beautiful',
 'beauty',
 'bestoftheday',
 'black',
 'blackandwhite',
 'blue',
 'cat',
 'catsofinstagram',
 'christmas',
 'cool',
 'cute',
 'daily',
 'design',
 'dog',
 'dogsofinstagram',
 'drawing',
 'explore',
 'family',
 'fashion',
 'fit',
 'fitness',
 'flowers',
 'food',
 'foodie',
 'foodporn',
 'friends',
 'fun',
 'funny',
 'girl',
 'girls',
 'gym',
 'hair',
 'handmade',
 'happiness',
 'happy',
 'healthy',
 'holiday',
 'home',
 'igers',
 'illustration',
 'inspiration',
 'instadaily',
 'instafashion',
 'instafood',
 'instagram',
 'instamood',
 'instapic',
 'japan',
 'landscape',
 'life',
 'lifestyle',
 'lol',
 'london',
 'love',
 'makeup',
 'me',
 'memes',
 'moda',
 'model',
 'motivation',
 'music',
 'nails',
 'nature',
 'night',
 'nofilter',
 'nyc',
 'ootd',
 'party',
 'photo',
 'photographer',
 'photography',
 'photooftheday',
 'picoftheday',
 'pink',
 'pretty',
 'puppy',
 'sea',
 'selfie',
 'sky',
 'smile',
 'sty

In [62]:
path = '/content/drive/MyDrive/인스타/hashtag/image/'
ll = {}
index = 0
for i in file_list:
  print('-'*40)
  print(i)
  image_list = os.listdir(path + i)
  for j in image_list:
    image = cv2.imread(path + i + '/' + j )
    filepath = path + 'all/'+ str(index) +'.jpg'
    cv2.imwrite(filepath, image)
    print(index)
    index += 1
  ll[i] = index - 1


스트리밍 출력 내용이 길어서 마지막 5000줄이 삭제되었습니다.
17675
17676
17677
17678
17679
17680
17681
17682
17683
17684
17685
17686
17687
17688
17689
17690
17691
17692
17693
17694
17695
17696
17697
17698
17699
17700
17701
17702
17703
17704
17705
17706
17707
17708
17709
17710
17711
17712
17713
17714
17715
17716
17717
17718
17719
17720
17721
17722
17723
17724
17725
17726
17727
17728
17729
17730
17731
17732
17733
17734
17735
17736
17737
17738
17739
17740
17741
17742
17743
17744
17745
17746
17747
17748
17749
17750
17751
17752
17753
17754
17755
17756
17757
17758
17759
17760
17761
17762
17763
17764
17765
17766
17767
17768
17769
17770
17771
17772
17773
17774
17775
17776
17777
17778
17779
17780
17781
17782
17783
17784
17785
17786
17787
17788
17789
17790
17791
17792
17793
17794
17795
17796
17797
17798
17799
17800
17801
17802
17803
17804
17805
17806
17807
17808
17809
17810
17811
17812
17813
17814
17815
17816
17817
17818
17819
17820
17821
17822
17823
17824
17825
17826
17827
17828
17829
17830
17831
17832
17833
17834
1783

amazing: 211 
architecture: 216
art: 305
artist: 247
baby: 186
beach: 322
beautiful: 272
beauty: 233
bestofday: 218
black: 273
blackandwhite: 245
blue: 244
cat: 229
catsofinstagram: 185
christmas: 205
cool: 224
cute: 268
daily: 44
design: 266
dog: 205
dogsofinstagram: 364
drawing: 238
explore: 147
family: 279
fashion: 251
fit: 208
fitness: 129
flowers: 244
food: 168
foodie: 241
foodporn: 215
friends: 277
fun: 269
funny: 106
girl: 150
girls: 164
gym: 186
hair: 265
handmade: 211
happiness: 268
happy: 220
healthy: 305
holiday: 286
home: 290
igers: 204
illustration: 242
inspiration: 173
instadaily: 245
instafashion: 196
instafood: 154
instagram: 267
instamood: 206
instapic: 212
japan: 84
landscape: 255
life: 255
lifestyle: 154
lol: 221
london: 65
love: 267
makeup: 212
me: 272
memes: 230
moda: 66
model: 275
motivation: 258
music: 202
nails: 151
nature: 296
night: 274
nofilter: 292
nyc: 301
ootd: 97
party: 259
photo: 184
photographer: 207
photography: 300
photooftheday: 255
picoftheday: 164
pink: 309
pretty: 292
puppy: 316
sea: 226
selfie: 152
sky: 156
smile: 265
style: 226
summer: 270
sunset: 293
swag: 215
tattoo: 220
travel: 279
vsco: 286
vscocam: 257
wedding: 245
weekend: 233
winter: 87
work: 301
workout: 150
yummy: 283
















In [54]:
ll

{'amazing': 210,
 'architecture': 426,
 'art': 731,
 'artist': 978,
 'baby': 1164,
 'beach': 1486,
 'beautiful': 1758,
 'beauty': 1991,
 'bestoftheday': 2209,
 'black': 2482,
 'blackandwhite': 2727,
 'blue': 2971,
 'cat': 3200,
 'catsofinstagram': 3385,
 'christmas': 3590,
 'cool': 3814,
 'cute': 4082,
 'daily': 4126,
 'design': 4392,
 'dog': 4597,
 'dogsofinstagram': 4961,
 'drawing': 5199,
 'explore': 5346,
 'family': 5625,
 'fashion': 5876,
 'fit': 6084,
 'fitness': 6213,
 'flowers': 6457,
 'food': 6625,
 'foodie': 6866,
 'foodporn': 7081,
 'friends': 7358,
 'fun': 7627,
 'funny': 7733,
 'girl': 7883,
 'girls': 8047,
 'gym': 8233,
 'hair': 8498,
 'handmade': 8709,
 'happiness': 8977,
 'happy': 9197,
 'healthy': 9502,
 'holiday': 9788,
 'home': 10078,
 'igers': 10282,
 'illustration': 10524,
 'inspiration': 10697,
 'instadaily': 10942,
 'instafashion': 11138,
 'instafood': 11292,
 'instagram': 11559,
 'instamood': 11765,
 'instapic': 11977,
 'japan': 12061,
 'landscape': 12316,
 'lif

In [51]:
path = '/content/drive/MyDrive/인스타/hashtag/all_4.csv'
all = pd.read_csv(path , index_col=0)
all

index                                           hashtags  \
0          1  ['happy', 'fashion', 'vestidoscortos', 'igers'...   
1          2  ['love', 'xtina', 'idol', 'best', 'queen', 'di...   
2          3  ['cateblanchett', 'cateblanchettmyqueen', 'cat...   
3          6  ['photographylovers', 'photooftheday', 'firstp...   
4          7  ['ubudfoodfestival', 'wisatakuliner', 'saturda...   
...      ...                                                ...   
22630    332  ['2tiercake', 'strawberrycake', 'trufflecake',...   
22631    333  ['yummy', 'healthyfood', 'breakfast', 'kuliner...   
22632    335  ['eatbooksg', 'foodsg', 'foodiesg', 'instafood...   
22633    336  ['foodporn', 'foodie', 'foodphotography', 'ins...   
22634    339  ['homecooking', 'foodie', 'foodporn', 'food', ...   

                                          after_contents  \
0      en esta foto me veo gordita pero aja a mi me g...   
1                                                credit    
2      she gave me her usual smile which stopped my b...   
3                                          light swords    
4         sabtu malam minggu enaknya wisata kuliner aja    
...                                                  ...   
22630                                       dm to order    
22631                               rosti smoked salmon    
22632  27 layers croissant from it s better than tion...   
22633  hummus pita chicken skewers tomato goat cheese...   
22634                           yet another greek night    

                                            new_hashtags  \
0      [16, 6, 63, 21, 29, 47, 81, 23, 31, 13, 113, 5...   
1           [0, 543, 617, 5, 598, 47, 410, 19, 275, 756]   
2                      [470, 0, 18, 253, 212, 21, 5, 47]   
3         [220, 4, 265, 2, 34, 672, 47, 73, 1, 465, 399]   
4                                               [95, 47]   
...                                                  ...   
22630                 [349, 725, 601, 95, 519, 240, 526]   
22631                                     [95, 196, 317]   
22632                   [108, 95, 520, 74, 83, 166, 169]   
22633          [74, 83, 149, 108, 95, 161, 321, 160, 31]   
22634  [83, 74, 40, 257, 248, 149, 108, 430, 169, 166...   

                                     new_hashtags_onehot filename  drop  
0      [0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, ...  amazing   0.0  
1      [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  amazing   4.0  
2      [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  amazing   5.0  
3      [0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  amazing   5.0  
4      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...  amazing   8.0  
...                                                  ...      ...   ...  
22630  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    yummy   NaN  
22631  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    yummy   NaN  
22632  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    yummy   NaN  
22633  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    yummy   NaN  
22634  [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...    yummy   NaN  

[22635 rows x 7 columns]

In [61]:
index = 0
for i in file_list:
  cnt= 0
  for j in range(len(all)):
    if all['filename'][j] == i :
      cnt += 1
  print(i)
  print(cnt)
  print('-'*20)


amazing
211
--------------------
architecture
216
--------------------
art
305
--------------------
artist
247
--------------------
baby
186
--------------------
beach
322
--------------------
beautiful
272
--------------------
beauty
233
--------------------
bestoftheday
218
--------------------
black
273
--------------------
blackandwhite
245
--------------------
blue
244
--------------------
cat
229
--------------------
catsofinstagram
185
--------------------
christmas
205
--------------------
cool
224
--------------------
cute
268
--------------------
daily
44
--------------------
design
266
--------------------
dog
205
--------------------
dogsofinstagram
364
--------------------
drawing
238
--------------------
explore
147
--------------------
family
279
--------------------
fashion
251
--------------------
fit
208
--------------------
fitness
129
--------------------
flowers
244
--------------------
food
168
--------------------
foodie
241
--------------------
foodporn
215
----

In [63]:
path = '/content/drive/MyDrive/인스타/hashtag/image/all'
file_list = os.listdir(path)
print(file_list)

['0.jpg', '1.jpg', '2.jpg', '3.jpg', '4.jpg', '5.jpg', '6.jpg', '7.jpg', '8.jpg', '9.jpg', '10.jpg', '11.jpg', '12.jpg', '13.jpg', '14.jpg', '15.jpg', '16.jpg', '17.jpg', '18.jpg', '19.jpg', '20.jpg', '21.jpg', '22.jpg', '23.jpg', '24.jpg', '25.jpg', '26.jpg', '27.jpg', '28.jpg', '29.jpg', '30.jpg', '31.jpg', '32.jpg', '33.jpg', '34.jpg', '35.jpg', '36.jpg', '37.jpg', '38.jpg', '39.jpg', '40.jpg', '41.jpg', '42.jpg', '43.jpg', '44.jpg', '45.jpg', '46.jpg', '47.jpg', '48.jpg', '49.jpg', '50.jpg', '51.jpg', '52.jpg', '53.jpg', '54.jpg', '55.jpg', '56.jpg', '57.jpg', '58.jpg', '59.jpg', '60.jpg', '61.jpg', '62.jpg', '63.jpg', '64.jpg', '65.jpg', '66.jpg', '67.jpg', '68.jpg', '69.jpg', '70.jpg', '71.jpg', '72.jpg', '73.jpg', '74.jpg', '75.jpg', '76.jpg', '77.jpg', '78.jpg', '79.jpg', '80.jpg', '81.jpg', '82.jpg', '83.jpg', '84.jpg', '85.jpg', '86.jpg', '87.jpg', '88.jpg', '89.jpg', '90.jpg', '91.jpg', '92.jpg', '93.jpg', '94.jpg', '95.jpg', '96.jpg', '97.jpg', '98.jpg', '99.jpg', '100.jpg'

In [64]:
len(file_list)

22635